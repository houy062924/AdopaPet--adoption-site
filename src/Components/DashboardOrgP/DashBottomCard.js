import React from "react";

class DashBottomCard extends React.Component {
  constructor(props) {
    super(props);

    this.openFullProfile = this.openFullProfile.bind(this);
  }

  openFullProfile(p, i) {
    this.props.functions.openFullProfile(p, i);
  }

  render() {
    return (
      <div 
        key={this.props.profile.id} 
        className="cardCont" 
        onClick={()=> {this.openFullProfile(this.props.profile, this.props.index)}}
      >
        <div className="profileImg">
          <img src={this.props.profile.url}></img>
        </div>
        
        <div className="textCont">
          <h1 className="profileName">
            { this.props.profile.name }
          </h1>
          <p className="profileId">
            <span className="labelText">ID<br></br></span>
            { this.props.profile.id }
          </p>
          <p className="profileDays">
            <span className="labelText">Date<br></br></span>
            { this.props.profile.date }
          </p>
          <p className="profileId">
            <span className="labelText">Gender<br></br></span>
            { this.props.profile.gender }
          </p>
          <p className="profileLocation">
            <span className="labelText">Location<br></br></span>
            { this.props.profile.address }
          </p>
        </div>
        
      
      </div>
    )
  }
}

export default DashBottomCard;