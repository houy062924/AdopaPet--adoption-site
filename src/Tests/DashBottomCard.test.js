import React from 'react';
import DashBottomCard from "../Components/DashboardOrgP/DashBottomCard";
import renderer from 'react-test-renderer';

test("Test DashBottomCard Component", ()=>{
  let profile = {
    address: "浪浪別哭(台南店)",
    adoptionstatus: 1,
    date: "2020-08-12",
    gender: "Male",
    id: "sKSHOps2CX27St0d4SKT",
    month: "2",
    name: "阿波",
    orgname: "Taipei Shelter",
    orguid: "xyofYzEJhOhEFObcTZbca1Tnhl03",
    story: "這次的新生，一口氣來了2個年紀只有快兩個月大↵體型比巴掌大一點的小奶娃！↵他們來自彰化的鄉下↵飼主因為沒有節育觀念而讓家中的母狗懷孕生了寶寶，↵愛媽出手救援並且替狗媽媽結了紥↵這二個孩子輾轉之下到了台北浪浪找家↵黑白色的小乳牛，是個膽子大、個性外向的小女孩↵遇到其他體型較大的孩子來邀玩，也勇於奮戰！↵白色北極熊寶寶，是個個性腼腆的男孩，↵對於孩子們的戰鬥，他寧可選擇站在一旁保持自己的優雅, ↵待小乳牛回來後再和她一起過二人世界。",
    timestamp: {seconds: 1596387516, nanoseconds: 955000000},
    url: "https://firebasestorage.googleapis.com/v0/b/adoption-platform.appspot.com/o/images%2F%E9%98%BF%E6%B3%A2.jpg?alt=media&token=5cfd43f4-c3f2-4473-b7b4-bd0011d5bbbf",
    year: "0",
  }
  
  let comp = renderer.create(<DashBottomCard profile={profile} />).toJSON();
	expect(comp).toMatchSnapshot();
	comp = renderer.create(<DashBottomCard profile={{}} />).toJSON();
	expect(comp).toMatchSnapshot();
});