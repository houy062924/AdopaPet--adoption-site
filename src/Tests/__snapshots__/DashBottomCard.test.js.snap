// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DashBottomCard Component 1`] = `
<div
  className="cardCont"
  onClick={[Function]}
>
  <div
    className="profileImg"
  >
    <img
      src="https://firebasestorage.googleapis.com/v0/b/adoption-platform.appspot.com/o/images%2F%E9%98%BF%E6%B3%A2.jpg?alt=media&token=5cfd43f4-c3f2-4473-b7b4-bd0011d5bbbf"
    />
  </div>
  <div
    className="textCont"
  >
    <h1
      className="profileName"
    >
      阿波
    </h1>
    <p
      className="profileId"
    >
      <span
        className="labelText"
      >
        ID
        <br />
      </span>
      sKSHOps2CX27St0d4SKT
    </p>
    <p
      className="profileDays"
    >
      <span
        className="labelText"
      >
        Date
        <br />
      </span>
      2020-08-12
    </p>
    <p
      className="profileId"
    >
      <span
        className="labelText"
      >
        Gender
        <br />
      </span>
      Male
    </p>
    <p
      className="profileLocation"
    >
      <span
        className="labelText"
      >
        Location
        <br />
      </span>
      浪浪別哭(台南店)
    </p>
  </div>
</div>
`;

exports[`Test DashBottomCard Component 2`] = `
<div
  className="cardCont"
  onClick={[Function]}
>
  <div
    className="profileImg"
  >
    <img />
  </div>
  <div
    className="textCont"
  >
    <h1
      className="profileName"
    />
    <p
      className="profileId"
    >
      <span
        className="labelText"
      >
        ID
        <br />
      </span>
    </p>
    <p
      className="profileDays"
    >
      <span
        className="labelText"
      >
        Date
        <br />
      </span>
    </p>
    <p
      className="profileId"
    >
      <span
        className="labelText"
      >
        Gender
        <br />
      </span>
    </p>
    <p
      className="profileLocation"
    >
      <span
        className="labelText"
      >
        Location
        <br />
      </span>
    </p>
  </div>
</div>
`;
